src/components/RegisterPanel.js

```diff
@@ -18,10 +19,61 @@ const RegisterPanel = () => {
     { ...allRegisters[i], value: reg.low_ }
   ));
 
+  const getFilteredRegisters = () => {
+    switch (filter) {
+      case 'callee-saved':
+        return registersAppended.filter((r) => r.saver === 'Callee');
+      case 'caller-saved':
+        return registersAppended.filter((r) => r.saver === 'Caller');
+      case 'all':
+        return registersAppended;
+      default:
+        throw Error('unknown filter');
+    }
+  };
+
+  const filteredRegisters = getFilteredRegisters();
+  const getClassName = (type) => (type === filter ? 'btn-active' : undefined);
+
+  const CurrentCategory = () => (
+    <div className="btn-group" role="group">
+      <button
```

Comment:
all the buttons are basically the same with the class names being different, we could possibly factor the button out to get more reuse